File: CPP/No119.cc
/**
 * Created by Xiaozhong on 2020/8/6.
 * Copyright (c) 2020/8/6 Xiaozhong. All rights reserved.
 */

#include "iostream"
#include "vector"

using namespace std;

class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> ans({1});
        if (!rowIndex) return ans;
        for (int i = 0; i <= rowIndex; i++) {
            vector<int> temp;
            for (int j = 0; j <= i; j++)
                if (j == 0 || j == i) temp.push_back(1); else temp.push_back(ans[j] + ans[j - 1]);
            ans = temp;
        }
        return ans;
    }
};

int main() {
    Solution s;
    vector<int> ans = s.getRow(5);
    for (int i : ans) {
        cout << i << " ";
    }
}

File: Java/src/No119.java
/*
 * Copyright (c) 9/4/2021, HuXiaozhong. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 */

import java.util.ArrayList;
import java.util.List;

public class No119 {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> ans = new ArrayList<>();
        ans.add(1);
        if (rowIndex == 0) return ans;
        for (int i = 1; i <= rowIndex; ++i) {
            List<Integer> buffer = new ArrayList<>();
            buffer.add(ans.get(0));
            for (int idx = 1; idx <= ans.size() - 1; ++idx) buffer.add(ans.get(idx) + ans.get(idx - 1));
            buffer.add(ans.get(ans.size() - 1));
            ans = buffer;
        }
        return ans;
    }

    public static void main(String[] args) {
        No119 no119 = new No119();
        System.out.println(no119.getRow(3));
    }
}
